### ISSUE TITLE
Nav (not nav item) blueprint/data

### ISSUE THREAD

**Issue by edalzell** [statamic/ideas#690]
Use Case:

- making a nav where each "type" needs to be handled different. So a nav itself needs some data.

Would be great to be able to attach this data to the nav via a blueprint.

Current workaround is to either make an assumption (leads to more fragile code), or making a top level item with some data, and therefore not used fields on the items.

Secondarily, need to be able to show/hide fields based on a Nav's `type`.

**Comment by jackmcdade** [comment]
This hurts my brain.

![ELI8](https://i.pinimg.com/originals/41/fe/07/41fe070cc7d85e52045c855acb66cd7f.png)

**Comment by edalzell** [comment]
We have what we call a "super" menu:
<img width="958" alt="CleanShot 2021-11-04 at 11 40 35@2x" src="https://user-images.githubusercontent.com/6069653/140400291-0f9d7b4b-b85a-4c07-8efe-506905c0975f.png">

Issue is that each of those menus needs to handled differently, for example one looks like this:
<img width="1232" alt="CleanShot 2021-11-04 at 11 42 05@2x" src="https://user-images.githubusercontent.com/6069653/140400492-ee6de422-4aca-44b5-b937-1c0ca805f7bf.png">

and another: 
<img width="1247" alt="CleanShot 2021-11-04 at 11 42 27@2x" src="https://user-images.githubusercontent.com/6069653/140400535-31b894c8-7042-4a20-b5c3-26d16b424d64.png">

and yet another:
<img width="995" alt="CleanShot 2021-11-04 at 11 42 46@2x" src="https://user-images.githubusercontent.com/6069653/140400588-2860c99b-c641-4759-b391-b4439c4a898d.png">

So each of these needs a `type` so we can pull in the correct partial. Not only that, some of those type have specific fields we want to show for the items (like an `asset` field on the ones w/ pictures), but NOT show on the other menu types.

So it would be loverly to be able to attach the blueprint to the nav itself and not the items, but have the item blueprint be aware of the nav type, so we can show/hide fields. Alternatively, if we could have multiple nav item blueprints, that would also work.

**Comment by jackmcdade** [comment]
![ELI5](https://stimulatingminds.co.uk/wp-content/uploads/2020/02/d47sqw45bzu01-e1582542632454-1000x537.jpg)

**Comment by jackmcdade** [comment]
Just kidding i get it now, thanks! <3

**Comment by FrittenKeeZ** [comment]
I need a lot more flexibility for navigations as well.
I need to be able to disable external links for some navigations, and others I need customized blueprints for both internal and external links.

In this example I had to remove the built-in connection to collection and completely overtake the blueprint, as I was fighting against the CP handling
```yaml
sections:
  main:
    display: Main
    fields:
      -
        handle: link_type
        field:
          type: radio
          display: Type
          options:
            intern: Intern
            extern: Extern
          inline: true
          validate:
            - required
      -
        handle: title
        field:
          type: text
          display: Title
          validate:
            - required
      -
        handle: intern
        field:
          type: entries
          display: Intern
          max_items: 1
          create: false
          collections:
            - hotels
          validate:
            - 'required_if:link_type,intern'
          if:
            link_type: intern
      -
        handle: url
        field:
          type: text
          display: Extern
          placeholder: 'https://varbergskusthotell.se/'
          validate:
            - 'required_if:link_type,extern'
            - sometimes
            - nullable
            - url
          if:
            link_type: extern
      -
        handle: region
        field:
          type: terms
          display: Region
          create: false
          max_items: 1
          mode: select
          taxonomies:
            - regions
          validate:
            - 'required_if:link_type,extern'
          if:
            link_type: extern
      -
        handle: city
        field: common.label
        config:
          display: City
          if:
            link_type: extern
      -
        handle: map
        field:
          type: assets
          display: Map
          allow_uploads: false
          container: main
          folder: /maps
          max_files: 1
          mode: grid
          restrict: true

```
